src/taipy/core/scenario/_scenario_manager.py

```diff
@@ -105,46 +105,55 @@ def _create(
         creation_date: Optional[datetime.datetime] = None,
         name: Optional[str] = None,
     ) -> Scenario:
+        _task_manager = _TaskManagerFactory._build_manager()
+        _data_manager = _DataManagerFactory._build_manager()
+
         scenario_id = Scenario._new_id(str(config.id))
         cycle = (
             _CycleManagerFactory._build_manager()._get_or_create(config.frequency, creation_date)
             if config.frequency
             else None
         )
-
-        pipelines = [
-            _PipelineManagerFactory._build_manager()._get_or_create(p_config, cycle.id if cycle else None, scenario_id)
-            for p_config in config.pipeline_configs
-        ]
+        cycle_id = cycle.id if cycle else None
+        tasks = (
+            _task_manager._bulk_get_or_create(config.task_configs, cycle_id, scenario_id) if config.task_configs else []
+        )
+        additional_data_nodes = (
+            _data_manager._bulk_get_or_create(config.additional_data_node_configs, cycle_id, scenario_id)
+            if config.additional_data_node_configs
+            else {}
+        )
 
         is_primary_scenario = len(cls._get_all_by_cycle(cycle)) == 0 if cycle else False
         props = config._properties.copy()
         if name:
             props["name"] = name
         version = cls._get_latest_version()
+
         scenario = Scenario(
             str(config.id),
-            pipelines,
+            set(tasks),
```

Comment:
Please create a new issue in the product backlog to make `_task_manager._bulk_get_or_create` and `_data_manager._bulk_get_or_create` methods return a set directly. This could avoid converting the list into set.